src/election_anomaly/user_interface/__init__.py

```diff
@@ -730,16 +786,24 @@ def get_runtime_parameters(keys, param_file=None):
 	if len(p) == 0:
 		raise FileNotFoundError
 
-	for k in keys:
+	for k in required_keys:
+		try:
+			d[k] = parser['election_anomaly'][k]
+		except KeyError:
+			missing_required_params['missing'].append(k)
+
+	if optional_keys is None:
```

Comment:
I'm still getting tripped up around here through the next ~10 lines. I'm not sure why we need to create an empty list if `optional_keys` is None. Could you do an `else` and remove the try/except? And it seems strange to me that we're storing a number of fields in a dictionary where the value is `None`. Seems like we could just check for the existence of the key in the dictionary after the params are grabbed.

Reply:
Was also trying to avoid extra nesting, per my understanding of a previous suggestion from a trusted collaborator.

Reply:
re: dictionary keys: d['aux_data_dir'] is used in `election_anomaly/__init__.py` lines 132ff. If 'aux_data_dir' might not be a key in `d` then we have to put an `if-else` statement in this definition, which seems uglier to me.

```
    def load_results(self):
        results_info = dbr.get_datafile_info(self.session, self.d['results_file_short'])
        ui.new_datafile(self.session, self.munger, self.d['results_file'],
            juris=self.juris, project_root=self.d['project_root'], 
            results_info=results_info,aux_data_dir=self.d['aux_data_dir'])
```

Reply:
Ok! It all makes sense now.